librz/il/theory_fbasic.c

```diff
@@ -0,0 +1,415 @@
+
+	RzILOpArgsFCastsfloat cast = op->op.fcast_sfloat;
+	RzBitVector *bv = rz_il_evaluate_bitv(vm, cast.bv);
+	RzFloatFormat format = cast.format;
+	RzFloatRMode mode = cast.mode;
+	RzFloat *ret = rz_float_cast_float(bv, format, mode);
+
+	rz_bv_free(bv);
+
+	*type = RZ_IL_TYPE_PURE_FLOAT;
+	return ret;
+}
+
+void *rz_il_handler_fconvert(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	rz_return_val_if_fail(vm && op && type, NULL);
+
+	RzILOpArgsFconvert convert = op->op.fconvert;
+	RzFloat *f = rz_il_evaluate_float(vm, convert.f);
+	RzFloatFormat format = convert.format;
+	RzFloatRMode mode = convert.mode;
+	RzFloat *ret = rz_float_convert(f, format, mode);
+
+	rz_float_free(f);
+
+	*type = RZ_IL_TYPE_PURE_FLOAT;
+	return ret;
+}
+
+void *rz_il_handler_frequal(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	rz_return_val_if_fail(vm && op && type, NULL);
+	RzILOpArgsFrequal frequal = op->op.frequal;
+	RzILBool *ret = rz_il_bool_new(frequal.x == frequal.y);
+
+	*type = RZ_IL_TYPE_PURE_BOOL;
+	return ret;
+}
+
+void *rz_il_handler_fsucc(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	rz_return_val_if_fail(vm && op && type, NULL);
+
+	RzILOpArgsFsucc fsucc = op->op.fsucc;
+	RzFloat *f = rz_il_evaluate_float(vm, fsucc.f);
+	RzFloat *ret = rz_float_succ(f);
+
+	rz_float_free(f);
+
+	*type = RZ_IL_TYPE_PURE_FLOAT;
+	return ret;
+}
+
+void *rz_il_handler_fpred(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	rz_return_val_if_fail(vm && op && type, NULL);
+
+	RzILOpArgsFpred fpred = op->op.fpred;
+	RzFloat *f = rz_il_evaluate_float(vm, fpred.f);
+	RzFloat *ret = rz_float_pred(f);
+
+	rz_float_free(f);
+
+	*type = RZ_IL_TYPE_PURE_FLOAT;
+	return ret;
+}
+
+void *rz_il_handler_forder(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	rz_return_val_if_fail(vm && op && type, NULL);
+
+	RzILOpArgsForder forder = op->op.forder;
+	RzFloat *x = rz_il_evaluate_float(vm, forder.x);
+	RzFloat *y = rz_il_evaluate_float(vm, forder.y);
+	RzILBool *order = rz_il_bool_new(rz_float_cmp(x, y) == -1);
+
+	rz_float_free(x);
+	rz_float_free(y);
+
+	*type = RZ_IL_TYPE_PURE_BOOL;
+	return order;
+}
+
+void *rz_il_handler_fround(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	rz_return_val_if_fail(vm && op && type, NULL);
+
+	RzILOpArgsFround fround = op->op.fround;
+	RzFloat *f = rz_il_evaluate_float(vm, fround.f);
+	RzFloatRMode mode = fround.rmode;
+	RzFloat *ret = rz_float_round_to_integral(f, mode);
+
+	rz_float_free(f);
+
+	*type = RZ_IL_TYPE_PURE_FLOAT;
+	return ret;
+}
+
+void *rz_il_handler_fsqrt(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	rz_return_val_if_fail(vm && op && type, NULL);
+	RzILOpArgsFsqrt sqrt = op->op.fsqrt;
+	RzFloat *n = rz_il_evaluate_float(vm, sqrt.f);
+	RzFloatRMode mode = sqrt.rmode;
+	RzFloat *ret = rz_float_sqrt(n, mode);
+
+	rz_float_free(n);
+
+	*type = RZ_IL_TYPE_PURE_FLOAT;
+	return ret;
+}
+
+void *rz_il_handler_frsqrt(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
+	// TODO : float todo unimplemented
```

Comment:
Maybe print a warning?

Reply:
going to use `unimplemented handler` for those operations.